src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -80,13 +83,15 @@ public enum CAType {
     }
 
     @Inject
-    public CertificateStore(Kernel kernel) throws IOException {
-        this.workPath = kernel.getNucleusPaths().workPath(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME);
+    public CertificateStore(Kernel kernel, DomainEvents eventEmitter) throws IOException {
+        this(kernel.getNucleusPaths().workPath(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME),
+                eventEmitter);
     }
 
     // For unit tests
```

Comment:
This is kind of bad, just having an API for unit tests

Reply:
Agree, and I've left similar comments in the past. Though in this case it's a pain to set up the kernel object to return the right work path :-/